File: maxhanna.client/src/services/datacontracts/file/file-comment.ts
import { Reaction } from "../reactions/reaction";
import { User } from "../user/user";
import { FileEntry } from "./file-entry";
import { Poll } from "../social/poll";

export class FileComment {
  id!: number;
  user!: User;
  commentText?: string;
  upvotes: number = 0;
  downvotes: number = 0;
  storyId?: number;
  commentId?: number;
  fileId?: number;
  userProfileId?: number;
  commentFiles?: FileEntry[];
  date?: Date;
  city?: string;
  country?: string;
  ip?: string;
  reactions?: Array<Reaction> | undefined;
  comments?: Array<FileComment> | undefined;
  // New: polls associated with this comment
  polls?: Array<Poll> | undefined;
  decrypted?: boolean | undefined;
}

File: maxhanna.Server/Controllers/DataContracts/Files/FileComment.cs
using maxhanna.Server.Controllers.DataContracts.Users;
using maxhanna.Server.Controllers.DataContracts.Social; // For Poll

namespace maxhanna.Server.Controllers.DataContracts.Files
{
	public class FileComment
	{
		public int Id { get; set; }
		public int? FileId { get; set; }
		public int? StoryId { get; set; }
		public int? CommentId { get; set; }
		public int? UserProfileId { get; set; }
		public string? City { get; set; }
		public string? Country { get; set; }
		public string? Ip { get; set; }
		public User? User { get; set; }
		public string? CommentText { get; set; }
		public DateTime Date { get; set; }
		public List<FileEntry>? CommentFiles { get; set; } = new List<FileEntry>();
		public List<FileComment>? Comments { get; set; } = new List<FileComment>();
		public List<Reaction>? Reactions { get; set; }
		// New: Polls associated with this comment (if the comment contains a poll markup or votes)
		public List<Poll>? Polls { get; set; }
	}
}
